### PR TITLE
documentation: fix minor misses from the remote function doc release

### PR DESCRIPTION
**Pull request by dpatro**
*Issue #, if available:*

*Description of changes:*
* Address some small misses from remote function doc release.

*Testing done:*
* ran `tox -e twine,sphinx --parallel` locally - no failures.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by knikure — COMMENTED** [review]
/bot run pr, slow-tests

File: src/sagemaker/remote_function/client.py [hunk 1]
```diff
@@ -87,7 +87,7 @@ def remote(
     This decorator wraps the annotated code and runs it as a new SageMaker job synchronously
     with the provided runtime settings.
 
-    Unless mentioned otherwise, the decorator first looks up the value from the SageMaker
+    In case a parameter value is not set, the decorator first looks up the value from the SageMaker
```
Comment (p-nanimal) [inline src/sagemaker/remote_function/client.py:90]: (If) a parameter value is not set, ....

File: src/sagemaker/remote_function/client.py [hunk 2]
```diff
@@ -475,10 +475,10 @@ def __init__(
     ):
         """Constructor for RemoteExecutor
 
-        Unless mentioned otherwise, the constructor first looks up the value from the SageMaker
-        configuration file. If no value is specified in the configuration file or no configuration
-        file is found, the constructor selects the default as specified below. For more
-        information, see `Configuring and using defaults with the SageMaker Python SDK
+        In case a parameter value is not set, the constructor first looks up the value from the
```
Comment (p-nanimal) [inline src/sagemaker/remote_function/client.py:478]: (If) a parameter value is....

File: src/sagemaker/remote_function/client.py [hunk 3]
```diff
@@ -1042,18 +1044,30 @@ def cancel(self):
             self._state = _CANCELLED
             return True
 
-    def running(self):
-        """Returns ``True`` if the underlying sagemaker job is still running."""
+    def running(self) -> bool:
+        """Check if the underlying sagemaker job is running.
```
Comment (p-nanimal) [inline src/sagemaker/remote_function/client.py:1048]: sagemaker->SageMaker

File: src/sagemaker/remote_function/client.py [hunk 4]
```diff
@@ -1042,18 +1044,30 @@ def cancel(self):
...
+
+        Returns:
+            ``True`` if the underlying sagemaker job is still running. ``False``, otherwise.
```
Comment (p-nanimal) [inline src/sagemaker/remote_function/client.py:1051]: sagemaker->SageMaker

File: src/sagemaker/remote_function/client.py [hunk 5]
```diff
@@ -1042,18 +1044,30 @@ def cancel(self):
...
+        """
         with self._condition:
             return self._state == _RUNNING
 
-    def cancelled(self):
-        """Returns ``True`` if the underlying sagemaker job was cancelled. ``False``, otherwise."""
+    def cancelled(self) -> bool:
+        """Check if the underlying sagemaker job was cancelled.
```
Comment (p-nanimal) [inline src/sagemaker/remote_function/client.py:1057]: search for all sagemaker->SageMaker

Comment (dpatro) [reply src/sagemaker/remote_function/client.py:1057]: Done

File: src/sagemaker/remote_function/client.py (hunk 4 above)
Comment (dpatro) [reply src/sagemaker/remote_function/client.py:1051]: Done

File: src/sagemaker/remote_function/client.py (hunk 3 above)
Comment (dpatro) [reply src/sagemaker/remote_function/client.py:1048]: Done

File: src/sagemaker/remote_function/client.py (hunk 2 above)
Comment (dpatro) [reply src/sagemaker/remote_function/client.py:478]: Done

File: src/sagemaker/remote_function/client.py (hunk 1 above)
Comment (dpatro) [reply src/sagemaker/remote_function/client.py:90]: Done

**Review by p-nanimal — APPROVED** [review]

**Pull request merged**